homeworks/week6/hw6.py

```diff
@@ -0,0 +1,49 @@
+from collections import Counter
+def getShortestSubstrOfAllChar(S, T):
+    '''get shortest substring of all char. If there are multiple substrings with same length, return the leftmost one.
+
+    Time Complexity: O(max(N, M)) where N = length of S and M = length of T (in case O(N) < O(M) the time complexity is strictly O(N), it does not depend on the size of the alphabet |A|).
+
+    Auxiliary Space: O(|A|) where |A| = the number of different characters in T'''
+    #first, build a set of all chars in T
+    setT = set(T)
+    #Two pointers at the ends of the substring
+    i, j = 0, 0
+    #A HashMap (key, val) where key = character, val = frequency
+    d = {}
+    #Keep track of current optima
+    optval, opt = float('inf'), None
+    while i < len(S) and j < len(S):
+        while j < len(S) and len(d) < len(setT): #not enough
```

Comment:
Your solution of using two moving indices, i, j while using a mapCount is optimal.

nit: The variable names can be used to improve code readability.